src/lib.rs

```diff
@@ -128,7 +128,7 @@ impl Config {
 ///
 /// - `filename` doesn't exist
 /// - for any reason `filename` could not be read (maybe no read permission)
-pub fn run(config: Config) -> Result<(), Box<Error>> {
+pub fn run(config: &Config) -> Result<(), Box<Error>> {
```

Comment:
Given you edited the function signature, update the documentation (line 116) to indicate `run` expects a reference to a Config, instead of a Config.

Reply:
Function documentation is the root of all evil